### PR TITLE
N lexers

### LINKED ISSUE #213: Port missing N lexer
Port missing `n` lexers to Chroma.

- NASM
- NCL
- Nemerle
- NSIS
- objdump-nasm
- nesC
- Nimrod
- Nit
- Notmuch
- NuSMV
- NumPy

### PR DESCRIPTION
**Pull request by gandarez**
This PR adds missing `n` package lexers.

- Ncl
- NSIS
- Nasm Objdump
- nesC
- Nimrod
- Nit
- Notmuch
- NuSMV
- NumPy

Closes https://github.com/wakatime/wakatime-cli/issues/213

### CODE REVIEW

File: lexers/n/nsis.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import (
+	. "github.com/alecthomas/chroma" // nolint
+	"github.com/alecthomas/chroma/lexers/internal"
+)
+
+// NSIS lexer.
+var NSIS = internal.Register(MustNewLexer(
+	&Config{
+		Name:      "NSIS",
+		Aliases:   []string{"nsis", "nsis", "nsh"},
```
Comment (dron22) [inline lexers/n/nsis.go]: small typo here:
```suggestion
		Aliases:   []string{"nsis", "nsi", "nsh"},
```

File: lexers/n/numpy.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+var NumPy = internal.Register(MustNewLexer(
+	&Config{
+		Name:    "NumPy",
+		Aliases: []string{"numpy"},
+	},
+	Rules{
+		"root": {},
+	},
+).SetAnalyser(func(text string) float32 {
+	matched, _ := shebang.MatchString(text, `pythonw?(3(\.\d)?)?`)
```
Comment (dron22) [inline lexers/n/numpy.go]: yeah, this if condition is very long. I like the approach with `matched` variable. Maybe we can even extend it, to fully remove this long line.
```suggestion
	hasPythonShebang, _ := shebang.MatchString(text, `pythonw?(3(\.\d)?)?`)
	...

	if (hasPythonShebang || containsImport) && (hasNumpyImport || hasFromNumpyImport) {
	    return 1.0
	}
```

File: lexers/n/numpy_test.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	"io/ioutil"
+	"testing"
+
+	"github.com/alecthomas/assert"
+	"github.com/alecthomas/chroma"
+	"github.com/alecthomas/chroma/lexers/n"
+)
+
+func TestNumPy_AnalyseText(t *testing.T) {
+	data, err := ioutil.ReadFile("testdata/numpy")
```
Comment (dron22) [inline lexers/n/numpy_test.go]: As the text analysis has a more complicated if condition, can we add additional test cases?

**Review by dron22 — CHANGES_REQUESTED** [review]

**Comment by gandarez** [comment]
@dron22 fixes applied and test cases added for NumPy.

File: lexers/n/notmuch_test.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/alecthomas/assert"
+	"github.com/alecthomas/chroma"
+	"github.com/alecthomas/chroma/lexers/n"
+)
+
+func TestNtmuch_AnalyseText(t *testing.T) {
```
Comment (dron22) [inline lexers/n/notmuch_test.go]: small typo
```suggestion
func TestNotmuch_AnalyseText(t *testing.T) {
```

**Review by dron22 — APPROVED** [review]
Alright great. Just one small typo in test function name left. Approving already :+1:

**Pull request merged**